RHDEVDOCS-6201: Content creation for GitOps 1.14.1 RN (#83610)
openshift/openshift-docs: modules/gitops-release-notes-1-14-1.adoc

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-14.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-14-1_{context}"]
= Release notes for {gitops-title} 1.14.1

{gitops-title} 1.14.1 is now available on {OCP} 4.13, 4.14, 4.15, 4.16, and 4.17.

[id="errata-updates-1-14.1_{context}"]
== Errata updates

[id="RHEA-XXXX:NNNN-gitops-1-14-1-security-update-advisory_{context}"]
=== RHEA-XXXX:NNNN - {gitops-title} 1.14.1 security update advisory

Issued: 2024-XX-YY

The list of security fixes that are included in this release are documented in the following advisory:

* link:https://access.redhat.com/errata/RHEA-XXXX:NNNN[RHEA-XXXX:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

[id="fixed-issues-1-14-1_{context}"]
== Fixed issues

> * Before this update, resource limits set on the operator container led to functional issues on clusters containing a large number of `Secrets` and `ConfigMaps`. As a result, the operator container exceeded its allocated memory limits. With this update, the resource limits set on the operator container are removed to enhance functionality. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]

Reviewer: The text doesn't fully capture the intended change. Could you please revise it to align more closely with the RN text from JIRA?

Author: I have made a change that hopefully addresses this comment, PTAL.

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-14.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-14-1_{context}"]
= Release notes for {gitops-title} 1.14.1

> {gitops-title} 1.14.1 is now available on {OCP} 4.13, 4.14, 4.15, 4.16, and 4.17.

Reviewer 2: ```suggestion
{gitops-title} 1.14.1 is now available on {OCP} 4.13, 4.14, 4.15 and 4.16.
```

Yet to confirm the support for 4.17

Author: Fixed.

[id="errata-updates-1-14.1_{context}"]
== Errata updates

[id="RHEA-XXXX:NNNN-gitops-1-14-1-security-update-advisory_{context}"]
=== RHEA-XXXX:NNNN - {gitops-title} 1.14.1 security update advisory

Issued: 2024-XX-YY

The list of security fixes that are included in this release are documented in the following advisory:

* link:https://access.redhat.com/errata/RHEA-XXXX:NNNN[RHEA-XXXX:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

> [id="fixed-issues-1-14-1_{context}"]

Reviewer 2: Shall we add RN for https://issues.redhat.com/browse/GITOPS-4234 and https://issues.redhat.com/browse/GITOPS-5640

cc: @keithchong @jparsai

Reviewer 3: For 4234, we discussed this in the standup. It will improve performance and there is no functional change, so lets not RN it.

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-14.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-14-1_{context}"]
= Release notes for {gitops-title} 1.14.1

> {gitops-title} 1.14.1 is now available on {OCP} 4.13, 4.14, 4.15, and 4.16.

Reviewer 2: ```suggestion
{gitops-title} 1.14.1 is now available on {OCP} 4.13, 4.14, 4.15, 4.16 and 4.17.
```

Also, add 4.17 in 1.14.0 RN

Author: Fixed.

----

[id="fixed-issues-1-14-1_{context}"]
== Fixed issues

* Before this update, resource limits were set on the operator container as a best practice, which resulted in functional issues on clusters containing a large number of `Secrets` and `ConfigMaps`. This issue arises due to the operator container requiring more memory than the permitted limits. With this update, the resource limits set on the manager container have been removed to minimize the impact on functionality. Efforts are focused to optimize memory consumption for future releases. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]

* Before this update, Argo CD could not obtain the appropriate TLS certificate for Helm OCI registries when the URL included a path or port number. With this update, a fix is introduced in upstream Argo CD to correctly parse the URL and return a valid certificate. link:https://issues.redhat.com/browse/GITOPS-5081[GITOPS-5081]

[id="breaking-change-1-14-1_{context}"]
== Breaking change

[id="additional-configurations-for-cluster-scoped-rollouts-instance-installation_{context}"]
=== Additional configurations for cluster-scoped rollouts instance installation

> * When you upgrade to {gitops-title} v1.14, if you want to create the cluster-scoped rollouts installation outside the default installation namespace, `openshift-gitops-operator`, you must host it in the `CLUSTER_SCOPED_ARGO_ROLLOUTS_NAMESPACES` environment variable of the `Subscription` resource. The {gitops-title} Operator does not support cluster-scoped rollouts installation if the namespace is not defined in the `CLUSTER_SCOPED_ARGO_ROLLOUTS_NAMESPACES` environment variable.

Reviewer 4: Defalut namespace is openshift-gitops

Author: Fixed.

+

> In previous versions of {gitops-title}, Argo Rollouts used the `NAMESPACE_SCOPED_ARGO_ROLLOUTS_NAMESPACES` environment variable of the `Subscription` resource to check if the cluster-scoped rollouts instance can host a cluster-scoped rollouts installation in the user-defined namespace.

Reviewer 4: We can rephrase this line
`if the cluster-scoped rollouts instance can host a cluster-scoped rollouts installation in the user-defined namespace.` => `if user can create a cluster-scoped rollouts instance in the user-defined namespace.`

Author: Fixed.

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-14.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-14-1_{context}"]
= Release notes for {gitops-title} 1.14.1

{gitops-title} 1.14.1 is now available on {OCP} 4.13, 4.14, 4.15, 4.16, and 4.17.

[id="errata-updates-1-14.1_{context}"]
== Errata updates

[id="RHEA-XXXX:NNNN-gitops-1-14-1-security-update-advisory_{context}"]
=== RHEA-XXXX:NNNN - {gitops-title} 1.14.1 security update advisory

Issued: 2024-10-29

> The list of security fixes that are included in this release are documented in the following advisory:

Author: Fixed.

* link:https://access.redhat.com/errata/RHEA-XXXX:NNNN[RHEA-XXXX:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

[id="fixed-issues-1-14-1_{context}"]
== Fixed issues

> * Before this update, resource limits were set on the operator container as a best practice, which resulted in functional issues on clusters containing a large number of `Secrets` and `ConfigMaps`. This issue arises due to the operator container requiring more memory than the permitted limits. With this update, the resource limits set on the manager container have been removed to minimize the impact on functionality. Efforts are focused to optimize memory consumption for future releases. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]

Reviewer 5: Make the "pre-update" wording all past tense.
```suggestion
* Before this update, resource limits were set on the operator container as a best practice, which resulted in functional issues on clusters containing a large number of `Secrets` and `ConfigMaps`. This issue arose due to the operator container requiring more memory than the permitted limits. With this update, the resource limits set on the manager container have been removed to minimize the impact on functionality. Efforts are focused to optimize memory consumption for future releases. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]
```

Reviewer 5: Actually, I like the wording in the bug better, with a few tweaks. Up to you, of course. 

>Before this update, resource limits were set on the operator container as a recommended practice. However, these limits caused functional issues on clusters with a large number of secrets and configmaps because the operator required more memory than allowed by these limits.

Author: Fixed.

* Before this update, Argo CD could not obtain the appropriate TLS certificate for Helm OCI registries when the URL included a path or port number. With this update, a fix is introduced in upstream Argo CD to correctly parse the URL and return a valid certificate. link:https://issues.redhat.com/browse/GITOPS-5081[GITOPS-5081]

> * Before this update, you could not log in to the Argo CD web console UI after a Red Hat OpenShift Service on AWS cluster, that includes a GitOps Operator and an Argo CD instance configured with Dex-based SSO, resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]

Reviewer 5: ```suggestion
* Before this update, you could not log in to the Argo CD web console UI on a Red Hat OpenShift Service on AWS cluster that includes a GitOps Operator and an Argo CD instance configured with Dex-based SSO, after the cluster resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]
```

Author: Fixed.

[id="breaking-change-1-14-1_{context}"]
== Breaking change

[id="additional-configurations-for-cluster-scoped-rollouts-instance-installation_{context}"]
=== Additional configurations for cluster-scoped rollouts instance installation

> * When you upgrade to {gitops-title} v1.14, if you want to create the cluster-scoped rollouts installation outside the default installation namespace, `openshift-gitops`, you must host it in the `CLUSTER_SCOPED_ARGO_ROLLOUTS_NAMESPACES` environment variable of the `Subscription` resource. The {gitops-title} Operator does not support cluster-scoped rollouts installation if the namespace is not defined in the `CLUSTER_SCOPED_ARGO_ROLLOUTS_NAMESPACES` environment variable.

Reviewer 5: > host it

I'm not 100% certain what that this means. Add the namespace to the `CLUSTER_SCOPED_ARGO_ROLLOUTS_NAMESPACES` environment variable?  
Maybe an example (is this right?)

```
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: openshift-gitops-operator
  namespace: openshift-operators
# ...
spec:
  config:
    env:apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: openshift-gitops-operator
  namespace: openshift-operators
# ...
spec:
  config:
    env:
    - name: CLUSTER_SCOPED_ARGO_ROLLOUTS_NAMESPACES
      value: openshift-gitops, <list of namespaces of cluster-scoped Argo CD instances>
    - name: ARGOCD_CLUSTER_CONFIG_NAMESPACES
      value: openshift-gitops, <list of namespaces of cluster-scoped Argo CD instances>
```

Author: Hello @mburke5678, If an example clears things, I have one written in a different PR that relates to this RN entry for the feature. An example under this [section](https://83242--ocpdocs-pr.netlify.app/openshift-gitops/latest/argo_rollouts/using-cluster-scoped-rollouts-instance-to-manage-rollouts-resources.html) is something that I am proposing.

Reviewer 5: @Dhruv-Soni11 Perfect. Thanks!

Author: Fixed.

// Module included in the following assembly:
//
// * release_notes/gitops-release-notes-1-14.adoc

:_mod-docs-content-type: REFERENCE

[id="gitops-release-notes-1-14-1_{context}"]
= Release notes for {gitops-title} 1.14.1

{gitops-title} 1.14.1 is now available on {OCP} 4.13, 4.14, 4.15, 4.16, and 4.17.

[id="errata-updates-1-14.1_{context}"]
== Errata updates

[id="RHEA-XXXX:NNNN-gitops-1-14-1-security-update-advisory_{context}"]
=== RHEA-XXXX:NNNN - {gitops-title} 1.14.1 security update advisory

Issued: 2024-10-29

The list of security fixes that are included in this release is documented in the following advisory:

* link:https://access.redhat.com/errata/RHEA-XXXX:NNNN[RHEA-XXXX:NNNN]

If you have installed the {gitops-title} Operator in the default namespace, run the following command to view the container images in this release:

[source,terminal]
----
$ oc describe deployment gitops-operator-controller-manager -n openshift-gitops-operator
----

[id="fixed-issues-1-14-1_{context}"]
== Fixed issues

> * Before this update, resource limits were set on the operator container as a recommended practice. However, these limits caused functional issues on clusters with a large number of secrets and configmaps because the operator required more memory than allowed by these limits. With this update, the resource limits set on the manager container have been removed to minimize the impact on functionality. Efforts are focused to optimize memory consumption for future releases. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]

Reviewer 5: ```suggestion
* Before this update, resource limits were set on the operator container as a recommended practice. However, these limits caused functional issues on clusters with a large number of secrets and config maps because the operator required more memory than allowed by these limits. With this update, the resource limits set on the manager container have been removed to minimize the impact on functionality. Efforts are focused to optimize memory consumption for future releases. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]
```

Author: Fixed.

> * Before this update, resource limits were set on the operator container as a recommended practice. However, these limits caused functional issues on clusters with a large number of secrets and config maps because the operator required more memory than allowed by these limits. With this update, the resource limits set on the manager container have been removed to minimize the impact on functionality. Efforts are focused to optimize memory consumption for future releases. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]

Reviewer 6: See this L34: "Before this update, resource limits were set on the **operator container** ..."

Operator container? I do not see this term anywhere in OCP docs.
Something is not right here for sure.

Can you please check with SME and rewrite this properly?

> * Before this update, resource limits were set on the operator container as a recommended practice. However, these limits caused functional issues on clusters with a large number of secrets and config maps because the operator required more memory than allowed by these limits. With this update, the resource limits set on the manager container have been removed to minimize the impact on functionality. Efforts are focused to optimize memory consumption for future releases. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]

Reviewer 6: ```suggestion
* Before this update, resource limits were set on the operator container as a recommended practice. However, these limits caused functional issues on clusters with a large number of secrets and config maps because the Operator required more memory than allowed by these limits. With this update, the resource limits set on the manager container have been removed to minimize the impact on functionality. Efforts are focused to optimize memory consumption for future releases. link:https://issues.redhat.com/browse/GITOPS-5665[GITOPS-5665]
```

As per the OCP guidelines - https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#operator-name-capitalization, the term "Operator" is always capitalized.

> * Before this update, Argo CD could not obtain the appropriate TLS certificate for Helm OCI registries when the URL included a path or port number. With this update, a fix is introduced in upstream Argo CD to correctly parse the URL and return a valid certificate. link:https://issues.redhat.com/browse/GITOPS-5081[GITOPS-5081]

Reviewer 6: ```suggestion
* Before this update, Argo CD could not obtain the appropriate transport layer security (TLS) certificate for Helm Open Container Initiative (OCI) registries when the URL included a path or port number. With this update, a fix is introduced in upstream Argo CD to correctly parse the URL and return a valid certificate. link:https://issues.redhat.com/browse/GITOPS-5081[GITOPS-5081]
```

Author: Fixed.

> * Before this update, you could not log in to the Argo CD web console UI on a Red Hat OpenShift Service on AWS cluster that includes a GitOps Operator and an Argo CD instance configured with Dex-based SSO, after the cluster resumed from hibernation. The login screen would display an error indicating an invalid redirect URI in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]

Reviewer 6: ```suggestion
* Before this update, you could not log in to the Argo CD web console UI on a Red Hat OpenShift Service on Amazon Web Services (AWS) cluster that includes a {gitops-shortname} Operator and an Argo CD instance configured with Dex-based SSO, after the cluster resumed from hibernation. The login screen would display an error indicating an invalid redirect Uniform Resource Identifier (URI) in the Dex configuration. This update fixes the issue by ensuring that the correct Dex redirect URL is updated in the Argo CD configuration whenever the Argo CD server route is modified. link:https://issues.redhat.com/browse/GITOPS-4358[GITOPS-4358]
```

Author: Fixed.
